modules/apps/client-extension/client-extension-test/src/testIntegration/java/com/liferay/client/extension/type/internal/manager/test/CETManagerImplTest.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.client.extension.type.internal.manager.test;
+
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.client.extension.type.manager.CETManager;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.test.log.LogCapture;
+import com.liferay.portal.test.log.LogEntry;
+import com.liferay.portal.test.log.LoggerTestUtil;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Evan Thibodeau
+ */
+@RunWith(Arquillian.class)
+public class CETManagerImplTest {
+
+	@Test
+	public void testGetCET() throws PortalException {
+		try (LogCapture logCapture = LoggerTestUtil.configureLog4JLogger(
+				_CLASS_NAME_CET_MANAGER_IMPL, LoggerTestUtil.WARN)) {
+
+			Assert.assertNull(
+				_cetManager.getCET(
+					TestPropsValues.getCompanyId(),
+					_NONEXISTENT_EXTERNAL_REFERENCE_CODE));
+
+			List<LogEntry> logEntries = logCapture.getLogEntries();
+
+			Assert.assertEquals(logEntries.toString(), 1, logEntries.size());
+
+			LogEntry logEntry = logEntries.get(0);
+
+			Assert.assertEquals(
+				"No CET found for external reference code " +
+					_NONEXISTENT_EXTERNAL_REFERENCE_CODE,
+				logEntry.getMessage());
+		}
+	}
+
+	@Rule
+	public final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	private static final String _CLASS_NAME_CET_MANAGER_IMPL =
+		"com.liferay.client.extension.type.internal.manager.CETManagerImpl";
```

Comment:
Should we get the class name using `CETManagerImpl.class.getName()`? just in case class could be moved or renamed

Reply:
Have not checked this, but I've found cases where SF does not allow to import classes from packages containint `internal` in the name.

Anyways, even if we could do that, rather than a test failure we would get a compilation error, which is just a different signal to review the test. So, I don't see any advantage in doing so

Reply:
Yeah, I tried this out and SF doesn't complain, but the test itself breaks due to an unresolved requirement: `Unresolved requirement: Import-Package: com.liferay.client.extension.type.internal.manager`